logstash-core/lib/logstash/instrument/periodic_poller/jvm.rb

```diff
@@ -22,31 +40,46 @@ def initialize(metric, options = {})
     end
 
     def collect
-      raw = JRMonitor.memory.generate      
+      raw = JRMonitor.memory.generate
       collect_heap_metrics(raw)
       collect_non_heap_metrics(raw)
       collect_pools_metrics(raw)
       collect_threads_metrics
       collect_process_metrics
+      collect_gc_stats
     end
 
     private
 
-    def collect_threads_metrics      
+    def collect_gc_stats
+      garbage_collectors = ManagementFactory.getGarbageCollectorMXBeans()
+
+      garbage_collectors.each do |collector|
+        name = GarbageCollectorName.get(collector.getName())
+        if name.nil?
+          logger.error("Unkown garbage collector name", :name => name)
+        else
+          metric.gauge([:jvm, :gc, :collectors, name], :collection_count, collector.getCollectionCount())
```

Comment:
blah, this is an annoying situation.  Aligning with ES will deviate from the logstash metrics style guide proposal: https://github.com/elastic/logstash/issues/5716#issuecomment-236640359

Let's stay aligned with ES for now. +1 on keeping this the same, I suppose. :|